app/models/journable/historic_active_record_relation.rb

```diff
@@ -0,0 +1,290 @@
+  end
+
+  private
+
+  # Switch the database table, e.g. from `work_packages` to `work_package_journals`.
+  #
+  def switch_to_journals_database_table(relation)
+    relation.instance_variable_set :@table, model.journal_class.arel_table
+    relation
+  end
+
+  # Modify there where clauses such that e.g. the work-packages table is substituted
+  # with the work-package-journals table.
+  #
+  # When the where clause contains the `id` column, use `journals.journable_id` instead.
+  #
+  def substitute_database_table_in_where_clause(relation)
+    relation.where_clause.instance_variable_get(:@predicates).each do |predicate|
+      substitute_database_table_in_predicate(predicate)
+    end
+    relation
+  end
+
+  def substitute_database_table_in_predicate(predicate)
+    case predicate
+    when String
+      gsub_table_names_in_sql_string!(predicate)
+    when Arel::Nodes::HomogeneousIn,
+         Arel::Nodes::In,
+         Arel::Nodes::NotIn,
+         Arel::Nodes::Equality,
+         Arel::Nodes::NotEqual,
+         Arel::Nodes::LessThan,
+         Arel::Nodes::LessThanOrEqual,
+         Arel::Nodes::GreaterThan,
+         Arel::Nodes::GreaterThanOrEqual
+      if predicate.left.relation == arel_table or predicate.left.relation == journal_class.arel_table
+        case predicate.left.name
+        when "id"
+          predicate.left.name = "journable_id"
+          predicate.left.relation = Journal.arel_table
+        when "updated_at"
+          predicate.left.name = "created_at"
+          predicate.left.relation = Journal.arel_table
+        when "created_at"
+          predicate.left = Arel::Nodes::SqlLiteral.new("\"journables\".\"created_at\"")
+        else
+          predicate.left.relation = journal_class.arel_table
+        end
+      end
+    when Arel::Nodes::Grouping
+      substitute_database_table_in_predicate(predicate.expr.left)
+      substitute_database_table_in_predicate(predicate.expr.right)
+    else
+      raise NotImplementedError, "FIXME A predicate of type #{predicate.class.name} is not handled, yet."
+    end
+  end
+
+  # Add a timestamp condition: Select the work package journals that are the
+  # current ones at the given timestamp.
+  #
+  def add_timestamp_condition(relation)
+    relation \
+        .joins("INNER JOIN \"journals\" ON \"journals\".\"data_type\" = '#{model.journal_class.name}'
+            AND \"journals\".\"data_id\" = \"#{model.journal_class.table_name}\".\"id\"".gsub("\n", "")) \
+        .merge(Journal.at_timestamp(timestamp))
+  end
+
+  # Join the journables table itself because we need to take the `created_at` attribute from that.
+  # The `created_at` column is not present in the `work_package_journals` table.
+  #
+  def add_join_on_journables_table_with_created_at_column(relation)
+    relation \
+        .joins("INNER JOIN (SELECT id, created_at FROM \"#{model.table_name}\") AS journables
+            ON \"journables\".\"id\" = \"journals\".\"journable_id\"".gsub("\n", ""))
```

Comment:
*Phone call 2023-02-02*:

- Retrieving the historic value for the `position` column is not possible because this column is not journalized.
- We initialize the columns that are missing in the journalized table as `NULL` so that no database error is raised
- When trying to `pluck` or to retrieve such a value, we log a warning ➝ https://github.com/opf/openproject/pull/11243/commits/41470cac7faebde554ebe98c7a1539adffef2bc9